pkg/dynamic/client.go

```diff
@@ -105,6 +109,14 @@ func (c *knDynamicClient) ListSourcesTypes() (*unstructured.UnstructuredList, er
 	return c.ListCRDs(options)
 }
 
+// ListChannelsTypes returns installed knative messaging CRDs
+func (c *knDynamicClient) ListChannelsTypes() (*unstructured.UnstructuredList, error) {
+	options := metav1.ListOptions{}
+	channelsLabels := labels.Set{channelLabelKey: channelLabelValue}
+	options.LabelSelector = channelsLabels.String()
+	return c.ListCRDs(options)
```

Comment:
This might become an issue if the user is not allowed to access CRDs on the cluster (like this is the case for OpenShift)

@navidshaikh had the same issue when implementing `kn source list-types` and as far as i remember the solution was to introduce a fallback to well-known types if CRDs cannot be lister because of authentication reasons. 'would love to see something similar here, too.

Reply:
@Kaustubh-pande  `knative.dev/pkg/errors` now have a type for detecting the forbidden error. If you get the forbidden error, you can fallback to list here the (only one: imc) eventing built-in channel types.